packages/vite/src/node/server/middlewares/static.ts

```diff
@@ -107,40 +107,40 @@ export function serveRawFsMiddleware(
   }
 }
 
-export function isFileAccessAllowed(
+export function isFileServingAllowed(
   url: string,
-  { root, strict }: Required<FileSystemServeOptions>
+  server: ViteDevServer
 ): boolean {
-  return !strict || normalizePath(url).startsWith(root + path.posix.sep)
+  if (!server.config.server.fsServe.strict) return true
```

Comment:
> Accessing restricted files now emits warning in console.
Should this line be removed if you want to have this?

In that case, looks like we need to have a wrapper so this one returns true for strict false but emits the warning https://github.com/vitejs/vite/blob/19dae997f91607424af2d0e159ae2570463bbcb3/packages/vite/src/node/server/transformRequest.ts#L78

Reply:
In transform request, we don't emit warning / error, if it's not allowed, we simply don't forward to (result in 404).

The 403 only work with `/@fs/` (`import '../out/of/root') will also be transformed to `@fs` by Vite). Which you can see the `ensureServingAccess` function

Reply:
This is how it works in strict true. But in strict false mode, we are allowing these fallbacks. Shouldn't we also emit a warning in this case? If not, when we change the default to true we could break user code that is wrongly accessing these not-allowed files.

Reply:
Should be solved.